client/src/app/pages/applications/applications-table-analyze/applications-table-analyze.tsx

```diff
@@ -660,91 +634,46 @@ export const ApplicationsTableAnalyze: React.FC = () => {
+          what:
+            applicationsToDelete.length > 1
+              ? t("terms.application(s)").toLowerCase()
+              : t("terms.application").toLowerCase(),
         })}
-        titleIconVariant="warning"
-        aria-label="Applications bulk delete"
-        aria-describedby="applications-bulk-delete"
-        onClose={() => closeBulkDeleteModal()}
-        showClose={true}
-        actions={[
-          <Button
-            key="delete"
-            variant="danger"
-            onClick={() => {
-              let ids: number[] = [];
-              if (applicationToBulkDelete) {
-                applicationToBulkDelete?.forEach((application) => {
-                  if (application.id) ids.push(application.id);
-                });
-                if (ids)
-                  bulkDeleteApplication({
-                    ids: ids,
-                  });
-              }
-              closeBulkDeleteModal();
-              selectAll(false);
-            }}
-          >
-            {t("actions.delete")}
-          </Button>,
-          <Button
-            key="cancel"
-            variant="link"
-            onClick={() => closeBulkDeleteModal()}
-          >
-            {t("actions.cancel")}
-          </Button>,
-        ]}
-      >
-        {`${t("dialog.message.applicationsBulkDelete")} ${t(
+        titleIconVariant={"warning"}
+        isOpen={applicationsToDelete.length > 0}
+        message={`${t("dialog.message.applicationsBulkDelete")} ${t(
           "dialog.message.delete"
         )}`}
-      </Modal>
-
-      {isConfirmDialogOpen && (
-        <ConfirmDialog
-          title={t("dialog.title.delete", {
-            what: t("terms.application").toLowerCase(),
-          })}
-          isOpen={true}
-          titleIconVariant={"warning"}
-          message={t("dialog.message.delete")}
-          confirmBtnVariant={ButtonVariant.danger}
-          confirmBtnLabel={t("actions.delete")}
-          cancelBtnLabel={t("actions.cancel")}
-          onCancel={() => setIsConfirmDialogOpen(false)}
-          onClose={() => setIsConfirmDialogOpen(false)}
-          onConfirm={() => {
-            if (applicationToDeleteId) {
-              deleteApplication({ id: applicationToDeleteId });
-              setApplicationToDeleteId(undefined);
-            }
-            setIsConfirmDialogOpen(false);
-          }}
-        />
-      )}
-
+        aria-label="Applications bulk delete"
+        confirmBtnVariant={ButtonVariant.danger}
+        confirmBtnLabel={t("actions.delete")}
+        cancelBtnLabel={t("actions.cancel")}
+        onCancel={() => setApplicationsToDelete([])}
+        onClose={() => setApplicationsToDelete([])}
+        onConfirm={() => {
+          let ids: number[] = [];
+          applicationsToDelete.forEach((application) => {
+            if (application.id) ids.push(application.id);
+          });
+          if (ids) bulkDeleteApplication({ ids: ids });
+          setApplicationsToDelete([]);
```

Comment:
```suggestion
          const ids = applicationsToDelete
          .filter(application=>application?.id)
          .map(application => application.id)
          
          //TODO should have some sort of loading state / error state before closing the modal just incase the bulk delete fails. Then we could move the setApplicatoinsToDelete to a success/error callback after the async operation rather than it being synchronous as it sits here. 
          if (ids) bulkDeleteApplication({ ids: ids });
          setApplicationsToDelete([]);
```

Reply:
@ibolton336, that's a good opportunity indeed.

Reply:
@ibolton336, good idea to move setApplicationsToDelete to the success/error callbacks.
We don't really need to handle loading or error state when deleting items, the notification whether the delete operation is successful or fails, by providing information to the user.